src/container/srv_target.c

```diff
@@ -222,6 +222,9 @@ cont_child_aggregate(struct ds_cont_child *cont, uint64_t *sleep)
 
 	D_DEBUG(DB_EPC, DF_CONT"[%d]: Aggregating finished\n",
 		DP_CONT(cont->sc_pool->spc_uuid, cont->sc_uuid), tgt_id);
+free:
+	if (snapshots != NULL && snapshots != cont->sc_snapshots)
```

Comment:
'snapshots != cont->sc_snapshots' can't be used to determine if the 'snapshots' should be freed here, since the 'sc_snapshots' could be changed while above aggregation yielding.

Reply:
ah, then probably we can not use sc_snapshots here.